osu.Game/Screens/Play/BreakTracker.cs

```diff
@@ -2,40 +2,37 @@
 // See the LICENCE file in the repository root for full licence text.
 
 using System.Collections.Generic;
+using System.Linq;
 using osu.Framework.Bindables;
 using osu.Framework.Graphics;
 using osu.Game.Beatmaps.Timing;
 using osu.Game.Rulesets.Scoring;
+using osu.Game.Utils;
 
 namespace osu.Game.Screens.Play
 {
     public class BreakTracker : Component
     {
         private readonly ScoreProcessor scoreProcessor;
-
         private readonly double gameplayStartTime;
 
+        private PeriodTracker breaks;
+
         /// <summary>
         /// Whether the gameplay is currently in a break.
         /// </summary>
         public IBindable<bool> IsBreakTime => isBreakTime;
 
-        protected int CurrentBreakIndex;
-
         private readonly BindableBool isBreakTime = new BindableBool();
 
-        private IReadOnlyList<BreakPeriod> breaks;
-
         public IReadOnlyList<BreakPeriod> Breaks
         {
-            get => breaks;
             set
             {
-                breaks = value;
-
-                // reset index in case the new breaks list is smaller than last one
                 isBreakTime.Value = false;
-                CurrentBreakIndex = 0;
+
+                breaks = new PeriodTracker(value.Where(b => b.HasEffect)
+                                                .Select(b => new Period(b.StartTime, b.EndTime - BreakOverlay.BREAK_FADE_DURATION)));
```

Comment:
Is this supposed to have changed, by the inclusion of the fade duration?

Reply:
This hasn't been changed, the old logic uses `BreakPeriod.Contains(time)` which checks by `time >= Start && time <= (End - BREAK_FADE_DURATION)`. With the `PeriodTracker` component, I wrote it as you can see. (add periods that `b.HasEffect` with exclusion of the fade duration at the end)

Reply:
I kind of mindlessly glanced over that, good spot. This has the potential to crash for "too short" breaks (because `Period` checks if `start < end`).

Reply:
In this case it's safe because of the `HasEffect` check before it.

Reply:
Yes I've thought about that the time I wrote this, `HasEffect` checks if the duration is longer than `650`, so no problem there. I was just hesitant a bit since this logic will actually fail if that check inside the property changes.

I guess it's fine for this exact case.

Reply:
`BREAK_FADE_DURATION` is explicitly tied to `MIN_BREAK_DURATION` so it should be pretty obvious something is going on there if it's ever changed.